### PR TITLE
Add 2 arg @inferred

### PR DESCRIPTION
**Pull request by fchorney**
Added `@inferred [AllowedType] f(x)` from https://github.com/JuliaLang/julia/pull/27516

### CODE REVIEW

File: src/Compat.jl [hunk 1]
```diff
@@ -405,6 +405,56 @@ if VERSION < v"1.5.0-DEV.681"
     Base.union(r::Base.OneTo, s::Base.OneTo) = Base.OneTo(max(r.stop,s.stop))
 end
 
+# https://github.com/JuliaLang/julia/pull/27516
+if VERSION < v"1.2.0-DEV.77"
+    import Test: @inferred
+    import Core.Compiler: typesubtract
```
Comment (omus) [inline src/Compat.jl:411]:
```suggestion
    using Core.Compiler: typesubtract
```

**Comment by omus** [comment]
I'll merge this in an hour unless there are objections.

**Comment by martinholters** [comment]
Maybe we can wait for #705 to have that in 3.12.0, too? Not that important, we can just release a 3.13.0 shortly after, so if you become impatient, feel free to merge.

**Comment by omus** [comment]
Just waiting on CI now

**Pull request merged**